src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,33 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exeption.ValidationExeption;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws ValidationExeption {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new ValidationExeption("The user with this login is already exist!");
+        }
+
```

Comment:
lets check if the login is not taken in the end
check if the data is valid first

Reply:
done